### ISSUE TITLE
Magic numbers that cause Kilosort2 to crash

### ISSUE THREAD

**Issue by dimokaramanlis** [MouseLand/Kilosort2#9]
Hi Kilosort2 team,

I am looking forward to using the new algorithm. I tried to run Kilosort2 with 252-electrode data, recorded at 10k sampling rate. I set the template length (nt0) to 25 in the config file. However, I encountered errors in different steps of the algorithm, and tried to pinpoint them.

1. For a small number of batches (I had 195), the indices are out of range. So maybe 1:200 can be 1:min([200 size(cc,1)]). https://github.com/MouseLand/Kilosort2/blob/627bf32d1c3e634b4a32136d15584a664851440a/preProcess/sort_by_rastermap.m#L10

2. Matrix multiplication doesn't work here, because wPCAd is 61x3, while the first dimension of dWU is nt0 (25). I think this happens because of 61 appearing as a number in multiple places, where it should be nt0.
https://github.com/MouseLand/Kilosort2/blob/627bf32d1c3e634b4a32136d15584a664851440a/mainLoop/learnAndSolve8b.m#L121

3. https://github.com/MouseLand/Kilosort2/blob/627bf32d1c3e634b4a32136d15584a664851440a/mainLoop/extractPCfromSnippets.m#L13

4. https://github.com/MouseLand/Kilosort2/blob/627bf32d1c3e634b4a32136d15584a664851440a/mainLoop/extractTemplatesfromSnippets.m#L13-L17

5. Below, except for 61, there's also 20 (should be nt0min?).
https://github.com/MouseLand/Kilosort2/blob/627bf32d1c3e634b4a32136d15584a664851440a/mainLoop/get_SpikeSample.m#L6-L10

6. From its value, nt0max seems to be the same as nt0min here, maybe a different name could help?
https://github.com/MouseLand/Kilosort2/blob/627bf32d1c3e634b4a32136d15584a664851440a/mainLoop/getKernels.m#L4

Good luck with adding the final touches, thank you for the great tool!

Best,
Dimos

**Comment by marius10p** [comment]
Thanks, I think I got all of them. It runs through now for me when I set ops.nt0 = 25, even if the time duration is short. I think the batch size ops.NT should also be set adaptively based on sampling rate, because it controls the timescale of drift correction. I'm going to leave this issue open until I fix that.

**Comment by dimokaramanlis** [comment]
It is now running through for me as well. Thanks for the quick reply.

When loading the results in phy, I noticed a large temporal shift of the templates relative to the spike times. 

I think the culprit is here:
https://github.com/MouseLand/Kilosort2/blob/827aca3f9f969f5c6177f6e8ee1247f8992f6f6e/postProcess/splitAllClusters.m#L175-L180

In your last changes, nt0min should be passed as a parameter in mexSVDsmall2 (tmax), but instead is set to 0 in the variable Params.  I got rid of the shift by setting 
Params     = double([0 Nfilt 0 0 size(rez.W,1) Nnearest Nrank 0 0 Nchan NchanNear ops.nt0min 0]);
